pkg/compact/clean.go

```diff
@@ -41,10 +41,11 @@ func BestEffortCleanAbortedPartialUploads(ctx context.Context, logger log.Logger
 		}
 
 		deleteAttempts.Inc()
-		if err := block.Delete(ctx, logger, bkt, id); err != nil {
+		if err := block.MarkForDeletion(ctx, logger, bkt, id); err != nil {
```

Comment:
Wait... if these are partial uploads then we don't need to go through the whole process of marking and only then deleting them since the readers have never loaded these blocks, right? So this seems wrong.

Reply:
Hmm, Currently we are using partial blocks in this file itself, so we can ideally delete the block since there is no reader for the block. 
According to our proposal, we want to ideally delete all blocks in the same fashion, by first marking them for deletion and then deleting them. Anyway if we use MarkForDeletion, the block would be eventually deleted by blocks_cleaner. I don't see any harm in using MarkForDeletion.

Reply:
Yes I think we covered that in our design - it make sense to do any of it - so for consistency reasons we might stick to mark for deletion. (: But happy to disscuss.